python/ray/tune/execution/trial_runner.py

```diff
@@ -65,16 +65,24 @@ def _find_newest_experiment_checkpoint(ckpt_dir) -> Optional[str]:
     return max(full_paths)
 
 
-def _load_trial_from_checkpoint(trial_cp: dict, stub: bool = False, **kwargs):
+def _load_trial_from_checkpoint(
+    trial_cp: dict, stub: bool = False, **overwrite_checkpoint_kwargs
+):
+    # Update trial checkpoint with kwargs passed in
     new_trial = Trial(
-        trial_cp["trainable_name"], stub=stub, _setup_default_resource=False, **kwargs
+        trial_cp["trainable_name"],
+        stub=stub,
+        _setup_default_resource=False,
     )
+    trial_cp.update(overwrite_checkpoint_kwargs)
```

Comment:
This is where I overwrite the trial checkpoint's `local_dir` to the new one, then move on to restoring the rest of the trial state, and also performing the checkpoint manager updating logic.